ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -843,261 +910,150 @@ mod tests {
         }
     }
 
-    pub struct UnverifiedFp31Values {
-        x_left: Vec<Fp31>,
-        x_right: Vec<Fp31>,
-        y_left: Vec<Fp31>,
-        y_right: Vec<Fp31>,
-        prss_left: Vec<Fp31>,
-        prss_right: Vec<Fp31>,
-        z_right: Vec<Fp31>,
-    }
-
-    impl DZKPBaseField for Fp31 {
-        type UnverifiedFieldValues = UnverifiedFp31Values;
-
-        fn convert(
-            x_left: &BitArray32,
-            x_right: &BitArray32,
-            y_left: &BitArray32,
-            y_right: &BitArray32,
-            prss_left: &BitArray32,
-            prss_right: &BitArray32,
-            z_right: &BitArray32,
-        ) -> Self::UnverifiedFieldValues {
-            UnverifiedFp31Values {
-                x_left: x_left
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-                x_right: x_right
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-                y_left: y_left
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-                y_right: y_right
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-                prss_left: prss_left
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-                prss_right: prss_right
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-                z_right: z_right
-                    .chunks(8)
-                    .map(|x| Fp31::truncate_from(x.load_le::<u8>()))
-                    .collect(),
-            }
+    fn generate_batch(check_resizing: bool, segment_size: usize) -> Batch {
+        if check_resizing {
+            let mut batch = Batch::new(1);
+            batch
+                .inner
+                .insert(Gate::default(), UnverifiedValuesStore::new(1));
+            batch.inner.get_mut(&Gate::default()).unwrap().offset = 4;
+            batch
+        } else {
```

Comment:
`check_resizing` is not defined in any comments. It's also not clear from the code what it means. It's a boolean argument which is the hardest to interpret. This logic is also pretty confusing. This needs to be improved in some fashion.